appcues/src/main/java/com/appcues/ui/StateMachineDirectory.kt

```diff
@@ -26,15 +25,6 @@ internal class AppcuesFrameStateMachineOwner(
     }
 }
 
-internal class ModalStateMachineOwner(override val stateMachine: StateMachine) : StateMachineOwning {
-
-    override val renderContext: RenderContext = Modal
-
-    override suspend fun reset() {
-        stateMachine.stop(true)
-    }
-}
-
 internal class StateMachineDirectory {
```

Comment:
add to cleanup() to something like 

```kotlin
fun cleanup() {
   stateMachines.value.cleanup()

   stateMachines.value.removeAll { it.isInvalid() }
}
```

Reply:
I updated this in the latest commit if you want to take a look and see what you think. I think it did clean things up nicely. I changed `cleanup()` to `onScreenChange()` to more closely reflect what it was used for. I did that because I found it confusing to have that interface have both `reset()` and `cleanup()` functions that really had two different meanings, but sounded very similar and generic.